### ISSUE TITLE
Make local builds one-step

### ISSUE THREAD

**Issue by andrewpbray** [stat20/course-materials#457]
Building the staff and live site locally require running pre-render scripts. For actual publishing of the public websites, all of the steps are encapsulated in the workflows. Local development is best done by running `quarto preview` (for the public site) and `quarto preview --profile staff-site` for the staff site. That won't, however, run the scripts such as `propagate-dates.r` and `write-toc.r`, so dates and navigation can be confusing.

We should change the default workflow from calls to `quarto preview` with calls to ...

- make
- an r function
- a .sh script that can be called both locally and by the ci?

Each of these should replicate what's found in the CI workflow.

@ryanlovett what are your thoughts on which of these approaches is best?

**Comment by ryanlovett** [comment]
Using `make` makes the most sense to me since it can contain rules that run R or bash scripts, and you can [design the rules](https://blog.djnavarro.net/posts/2023-06-30_makefiles/#the-fourth-wall-shatters-into-little-shards-of-recursion) ([pandoc example](https://github.com/infologie/pandoc-ssp/blob/main/Makefile)) to ensure target output files are created. It would need to be added to the environment.

R or bash scripts would work fine as well. You would need to manually check for whether the scripts created the output files. I would only favor R over bash if the execution required data structures like hash tables.

**Comment by andrewpbray** [comment]
Got it. So could we replace the bash scripting in the workflows (the calls to pre-render r scripts and the call to `quarto publish`) with just a single call to make? That way we'd ensure the same workflow on CI and locally. I haven't worked with `make` before, but I can probably figure it out using that blog post and docs online.

The blog post, btw, is *excellent*.

**Comment by ryanlovett** [comment]
Exactly, the workflow could be replaced with a call to `make`. There would be one `all` target which encapsulates the whole build by depending on discrete targets, and those run the existing scripts. Fernando actually goes over `make` as part of 159. ([notes](https://ucb-stat-159-s23.github.io/site/lectures/automation/make.html); in the past he linked to [software carpentry](https://swcarpentry.github.io/make-novice/))
